src/auth.js

```diff
@@ -0,0 +1,24 @@
+const { OAuth2Client } = require('google-auth-library');
+
+const CLIENT_ID =
+  '270040816063-djog7f8mpvt4m162ak4n04bjmftg1bhc.apps.googleusercontent.com';
+const client = new OAuth2Client(CLIENT_ID);
+
+const requireAuth = async (ctx, next) => {
+  const [_, idToken] = ctx.header.authorization.split(' ');
```

Comment:
Could be worth adding a checks for:
 - Whether the auth header exists
 - Whether the `idToken` portion of the `split` has a value or not.

Currently, If you attempt to hit the `protected` endpoint without any headers, it'll throw a `500`.